src/AiSystem.cpp

```diff
@@ -2,59 +2,149 @@
 #include <time.h> /* time */
 
 #include "SDL.h"
+#include <glm/gtx/vector_angle.hpp>
 
 #include "AiSystem.h"
 
 AiSystem::AiSystem()
     : num_ai_(0)
-    , counter_(0) {
+    , counter_(0)
+    , whos_it_(0) {
     srand(time(NULL)); // AI uses randomness, so initialize random seed
     add_event_handler(EventType::ACTIVATE_AI, &AiSystem::handle_activate_ai, this);
+    add_event_handler(EventType::OBJECT_TRANSFORM_EVENT, &AiSystem::handle_object_transform, this);
 }
 
 void AiSystem::update() {
     // only update every 10 frames
-    if (++counter_ != 10) {
-        return;
+    if (++counter_ == 10) {
+        counter_ = 0;
     }
 
-    counter_ = 0;
-
-    for (int i = 0; i < num_ai_; i++) {
-        unsigned int random = rand() % 20;
-
-        if (random == 0) {
-            EventSystem::queue_event(
-                Event(
-                    EventType::KEYPRESS_EVENT,
-                    "player_id", 3 - i,
-                    "key", SDL_CONTROLLER_AXIS_LEFTX,
-                    "value", -32768
-                )
-            );
-        } else if (random == 1) {
-            EventSystem::queue_event(
-                Event(
-                    EventType::KEYPRESS_EVENT,
-                    "player_id", 3 - i,
-                    "key", SDL_CONTROLLER_AXIS_LEFTX,
-                    "value", 32768
-                )
-            );
-        } else {
-            EventSystem::queue_event(
-                Event(
-                    EventType::KEYPRESS_EVENT,
-                    "player_id", 3 - i,
-                    "key", SDL_CONTROLLER_AXIS_TRIGGERRIGHT,
-                    "value", rand() % 32768
-                )
-            );
+
+    for (int i = 3; i >= 4 - num_ai_; i--) {
+        if (i != whos_it_) {
+            path_to(i, cars_[whos_it_].car_transform_.position_);
+        } else if (counter_ == 0) {
```

Comment:
shoudln't this just be else?

Reply:
If they're it, we only want to change their random controlling every 10 frames.